op-challenger/game/fault/register.go

```diff
@@ -0,0 +1,74 @@
+package fault
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ethereum-optimism/optimism/op-challenger/config"
+	"github.com/ethereum-optimism/optimism/op-challenger/game/fault/trace/alphabet"
+	"github.com/ethereum-optimism/optimism/op-challenger/game/fault/trace/cannon"
+	faultTypes "github.com/ethereum-optimism/optimism/op-challenger/game/fault/types"
+	"github.com/ethereum-optimism/optimism/op-challenger/game/scheduler"
+	"github.com/ethereum-optimism/optimism/op-challenger/game/types"
+	"github.com/ethereum-optimism/optimism/op-challenger/metrics"
+	"github.com/ethereum-optimism/optimism/op-service/txmgr"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+var (
+	cannonGameType   = uint8(0)
+	alphabetGameType = uint8(255)
+)
```

Comment:
Should these game types be fault-specific (since this is in the `fault` package)?

Reply:
My current thinking is that each type of game knows how to register its variants.  I could see a centralised list of game type codes but I'm not sure where that would live currently, so it made sense to just keep it with the registration code for now.  The canonical list of these game types is actually in the contract deployment code since that's what defines the "source of truth".